tracer/test/benchmarks/Benchmarks.Trace/AppSecBodyBenchmark.cs

```diff
@@ -0,0 +1,247 @@
+{
+    [MemoryDiagnoser]
+    public class AppSecBodyBenchmark
+    {
+        private Security security;
+        private readonly ComplexModel complexModel = new()
+        {
+            Age = 12,
+            Gender = "Female",
+            Name = "Tata",
+            LastName = "Toto",
+            Address = new Address
+            {
+                Number = 12,
+                City = new City { Name = "Paris", Country = new Country { Name = "France", Continent = new Continent { Name = "Europe", Planet = new Planet { Name = "Earth" } } } },
+                IsHouse = false,
+                NameStreet = "lorem ipsum dolor sit amet"
+            },
+            Address2 = new Address
+            {
+                Number = 15,
+                City = new City
+                {
+                    Name = "Madrid",
+                    Country = new Country
+                    {
+                        Name = "Spain",
+                        Continent = new Continent
+                        {
+                            Name = "Europe",
+                            Planet = new Planet { Name = "Earth" }
+                        }
+                    }
+                },
+                IsHouse = true,
+                NameStreet = "lorem ipsum dolor sit amet"
+            },
+            Dogs = new List<Dog> {
+                    new Dog { Name = "toto", Dogs = new List<Dog> { new Dog { Name = "titi" }, new Dog { Name = "titi" } } },
+                    new Dog { Name = "toto", Dogs = new List<Dog> { new Dog { Name = "tata" }, new Dog { Name = "tata" } } },
+                    new Dog { Name = "tata", Dogs = new List<Dog> { new Dog { Name = "titi" }, new Dog { Name = "titi" }, new Dog { Name = "tutu" } } }
+                    }
+        };
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            security = Security.Instance;
+        }
+
+        [Benchmark]
+        public void AllCycleSimpleBody()
+        {
+#if NETFRAMEWORK
+            Debugger.Break();
+            using var ms = new MemoryStream();
+            using var sw = new StreamWriter(ms);
+            var httpContextMock = new HttpContext(new HttpRequest(string.Empty, string.Empty, string.Empty), new HttpResponse(sw));
+            security.InstrumentationGateway.RaiseBodyAvailable(httpContextMock, new Span(new SpanContext(1, 1), DateTimeOffset.UtcNow), new { });
+#else
+            var httpContextMock = new HttpContextMock(new HttpResponseMock());
+            security.InstrumentationGateway.RaiseBodyAvailable(httpContextMock, new Span(new SpanContext(1, 1), DateTimeOffset.UtcNow), new { });
+#endif
+        }
+
+        [Benchmark]
+        public void AllCycleMoreComplexBody()
+        {
+#if NETFRAMEWORK
+            using var ms = new MemoryStream();
+            using var sw = new StreamWriter(ms);
+            var httpContextMock = new HttpContext(new HttpRequest(string.Empty, string.Empty, string.Empty), new HttpResponse(sw));
```

Comment:
I wonder if this setup should be outside the benchmark itself? Probably not necessary though.

Reply:
thanks! Fixed  in 8eda4fec8c9b51c954871e7bc66197dfb77ec7d9, resources are then disposed after the execution